jscomp/ext/ext_char.ml

```diff
@@ -38,6 +38,12 @@ let valid_hex x =
   | 'A' .. 'F' -> true
   | _ -> false 
 
+let hex_value c =
+  match c with
+  | '0'..'9' -> (Char.code c) - 48
+  | 'a'..'f' -> (Char.code c) - (Char.code 'a') + 10
+  | 'A'..'F' -> (Char.code c) + 32 - (Char.code 'a') + 10
+  | _ -> 16 (* larger than any legal value *)
```

Comment:
shall we raise an exception here?
otherwise this seems not to be a generally useful function so we move it to the encoding module?

Reply:
Moved this inline inside `unicode_codepoint_escape` as we call it there only on valid hex symbols.